framework/src/main/java/org/checkerframework/common/accumulation/AccumulationTransfer.java

```diff
@@ -0,0 +1,161 @@
+package org.checkerframework.common.accumulation;
+
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+import java.util.List;
+import javax.lang.model.element.AnnotationMirror;
+import org.checkerframework.common.value.ValueCheckerUtils;
+import org.checkerframework.dataflow.analysis.FlowExpressions;
+import org.checkerframework.dataflow.analysis.FlowExpressions.Receiver;
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.framework.flow.CFAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+
+/**
+ * The default transfer function for an accumulation checker.
+ *
+ * <p>Subclasses should call the {@link #accumulate(Node, TransferResult, String...)} accumulate} or
+ * {@link #accumulate(MethodInvocationNode, TransferResult, String...)} methods to accumulate a
+ * string at a particular program point.
+ */
+public class AccumulationTransfer extends CFTransfer {
+
+    /** The type factory. */
+    protected final AccumulationAnnotatedTypeFactory typeFactory;
+
+    /**
+     * Required constructor.
+     *
+     * @param analysis the analysis
+     */
+    public AccumulationTransfer(CFAnalysis analysis) {
+        super(analysis);
+        typeFactory = (AccumulationAnnotatedTypeFactory) analysis.getTypeFactory();
+    }
+
+    /**
+     * Updates the estimate of how many things {@code node} has accumulated.
+     *
+     * @param node the node whose estimate should be expanded
+     * @param result the transfer result containing the store to be modified
+     * @param values the new accumulation values
+     */
+    public void accumulate(Node node, TransferResult<CFValue, CFStore> result, String... values) {
+        AnnotatedTypeMirror oldType = typeFactory.getAnnotatedType(node.getTree());
+        AnnotationMirror newAnno = getNewAnno(oldType, values);
+        insertIntoStores(result, node, newAnno);
+    }
+
+    /**
+     * Updates the estimate for the receiver and any other receiver-returning methods in a chain
+     * with this MethodInvocationNode to include the new values.
+     *
+     * <p>For example, if the argument is the expression {@code a.b().c()}, the new value is "foo",
+     * and b and c return their receiver (and are deterministic), all of the expressions {@code a},
+     * {@code a.b()}, and {@code a.b().c()} would have their estimates updated to include "foo". If
+     * any method in the chain is non-deterministic, its estimate will not be updated (but the rest
+     * of the chain is not affected).
```

Comment:
Does "not affected" mean they are not updated, or they are updated as if all the methods had been deterministic, or they are updated (not "not updated"), but maybe in a different way than if all the methods had been deterministic?